### PR TITLE
Added support for executeFenced for v1.3

### PR DESCRIPTION
**Pull request by vbenni**
Supports executeFenced interface.

Summary:
VTS
cros_nnapi_vts_1_3 --gtest_filter=*Fenced*:*FENCED*:-*conv*:*mobilenet*:*l2_norm*:*resize_nearest*
[----------] Global test environment tear-down
[==========] 5362 tests from 2 test suites ran. (22300 ms total)
[  PASSED  ] 2041 tests.
[  SKIPPED ] 3321 tests

CTS
cros_nnapi_cts --gtest_filter=*Fenced*:*FENCED*:-*conv*:*mobilenet*:*l2_norm*:*resize_nearest*:*svdf_bias_present*
[----------] Global test environment tear-down
[==========] 5409 tests from 3 test suites ran. (71612 ms total)
[  PASSED  ] 5409 tests.

### CODE REVIEW

File: BasePreparedModel.cpp [hunk 1]
```diff
@@ -424,12 +426,145 @@ Return<V1_3::ErrorStatus> BasePreparedModel::execute_1_3(
...
-    const V1_3::Request& request, const hidl_vec<hidl_handle>& waitFor, V1_2::MeasureTiming measure,
-    const V1_3::OptionalTimePoint& deadline,
+    const V1_3::Request& request1_3, const hidl_vec<hidl_handle>& waitFor,
+    V1_2::MeasureTiming measure, const V1_3::OptionalTimePoint& deadline,
     const V1_3::OptionalTimeoutDuration& loopTimeoutDuration,
     const V1_3::OptionalTimeoutDuration& duration, executeFenced_cb cb) {
     ALOGV("Entering %s", __func__);
-    // TODO: Add support
+
+    time_point driverStart, driverEnd, deviceStart, deviceEnd, driverAfterFence;
```
Comment (anikulk) [inline BasePreparedModel.cpp:435]: Nit : https://google.github.io/styleguide/cppguide.html#Local_Variables, recommends to declare vars as we use. we can follow this going forward.

File: BasePreparedModel.cpp [hunk 2]
```diff
@@ -424,12 +426,145 @@ Return<V1_3::ErrorStatus> BasePreparedModel::execute_1_3(
...
+    // rest of the interfaces are based on 1.0 request
+    auto request = convertToV1_0(request1_3);
+
+    // Wait for the dependent events to signal
+    for (const auto& fenceHandle : waitFor) {
+        if (!fenceHandle.getNativeHandle()) {
+            cb(V1_3::ErrorStatus::INVALID_ARGUMENT, hidl_handle(nullptr), nullptr);
+            return Void();
+        }
+        int syncFenceFd = fenceHandle.getNativeHandle()->data[0];
```
Comment (anikulk) [inline BasePreparedModel.cpp:458]: can syncFenceFd be const?

File: BasePreparedModel.cpp [hunk 3]
```diff
@@ -424,12 +426,145 @@ Return<V1_3::ErrorStatus> BasePreparedModel::execute_1_3(
...
+        ALOGE("%s Exception !!! %s", __func__, ex.what());
+        cb(V1_3::ErrorStatus::GENERAL_FAILURE, hidl_handle(nullptr), nullptr);
+        return Void();
+    }
+    if (measure == MeasureTiming::YES) deviceEnd = now();
+
+    for (size_t i = 0; i < request.outputs.size(); i++) {
+        auto outIndex = mModelInfo->getModelOutputIndex(i);
+        ALOGI("OutputIndex: %d", outIndex);
+        const std::string& outputNodeName = mNgc->getNodeName(outIndex);
```
Comment (anikulk) [inline BasePreparedModel.cpp:497]: mNgc isn't a very meaningful var name. I know its not added by you, Can we change this to mNgraphNetCreator?

**Review by anikulk — COMMENTED** [review]
Can you add more details to the commit message,
In addition to the top level comment usually we add details below 
Also include a comment regarding testing - below is syntax.  
"TEST :" and the test you ran

**Review by anikulk — APPROVED** [review]

**Review by Rjasuja — APPROVED** [review]
LGTM

**Comment by Nagamani71** [comment]
LGTM

**Pull request merged**